projects/packages/my-jetpack/src/products/class-product.php

```diff
@@ -455,7 +479,7 @@ public static function get_status() {
 			if ( static::$requires_site_connection && ! ( new Connection_Manager() )->is_connected() ) {
 				// Site has never been connected before
 				if ( ! Jetpack_Options::get_option( 'id' ) ) {
-					$status = Products::STATUS_NEEDS_FIRST_SITE_CONNECTION;
+					$status = static::is_owned() ? Products::STATUS_SITE_CONNECTION_ERROR : Products::STATUS_NEEDS_FIRST_SITE_CONNECTION;
```

Comment:
I see what you mean about the conditionals getting insane here 😅 

You could potentially add `&& !static::is_owned` to the if statement above to remove the ternary if here - slightly cleaner.